java/core/src/main/java/sleeper/core/statestore/FileReferenceStore.java

```diff
@@ -136,12 +136,25 @@ void atomicallyReplaceFileReferencesWithNewOne(String jobId, String partitionId,
      * Atomically updates the job field of file references, as long as the job field is currently unset. This will be
      * used for compaction job input files.
      *
-     * @param jobId          The job id which will be added to the {@link AllReferencesToAFile}
-     * @param fileReferences The {@link AllReferencesToAFile} whose status will be updated
+     * @param requests A list of {@link AssignJobIdRequest}s which should each be applied atomically
      * @throws FileReferenceNotFoundException      if a reference does not exist
      * @throws FileReferenceAssignedToJobException if a reference is already assigned to a job
      * @throws StateStoreException                 if the update fails for another reason
      */
+    default void atomicallyAssignJobIdsToFileReferences(List<AssignJobIdRequest> requests)
```

Comment:
Is there a better name for this method? It sounds like multiple job ids are being assigned to the same file references. It's also not clear if "atomically" in the method name and in the Javadoc means that the entire list will either be applied or not, or whether each job will be atomically applied so the first few might succeed and the rest might fail. I think we mean the latter.

Should the method just be called `assignJobIds`?

Reply:
Yes, I think so. I suppose this is a case where added complexity means that trying to name more descriptively can be misleading, because you can't capture everything that's happening.